File: covilo-core/src/main/java/me/kvdpxne/covilo/domain/persistence/CommentRepository.java
package me.kvdpxne.covilo.domain.persistence;

import java.util.Optional;
import java.util.UUID;
import java.util.stream.Stream;
import me.kvdpxne.covilo.domain.model.Comment;

public interface CommentRepository {

  Stream<Comment> getAll();
  Stream<Comment> getAllByUserIdentifier(final UUID identifier);

  Optional<Comment> getByIdentifier(final UUID identifier);

  void insert(final Comment comment);

  void update(final Comment comment);

  void delete(final Comment comment);

  long count();
}

File: covilo-core/src/main/kotlin/me/kvdpxne/covilo/domain/persistence/CommentRepository.kt
package me.kvdpxne.covilo.domain.persistence

import me.kvdpxne.covilo.domain.model.Comment
import me.kvdpxne.covilo.domain.model.Comments
import java.util.UUID

/**
 * @see Comment
 */
interface CommentRepository {

  fun findByIdentifier(identifier: UUID): Comment?
  fun findAll(): Comments

  fun insert(comment: Comment)
  fun update(comment: Comment)

  fun delete(identifier: UUID)
  fun deleteAll()

  fun count(): Long
}